modules/slms_3/1.0/slms_3.smk

```diff
@@ -0,0 +1,591 @@
+
+    placeholders = {
+        "REPODIR": repodir,
+        "MODSDIR": modsdir,
+        "SCRIPTSDIR": scriptsdir,
+    }
+    mconfig = op.walk_through_dict(mconfig, update_placeholders, **placeholders)
+
+    # Validate samples data frame
+    schemas_dir = os.path.join(modsdir, "schemas")
+    schemas = os.listdir(schemas_dir)
+    for schema in schemas:
+        snakemake.utils.validate(msamples, schema=os.path.join(schemas_dir, schema))
+
+    # Configure output directory if not specified and create it
+    if mconfig["dirs"].get("_parent") is None:
+        root_output_dir = mconfig.get("root_output_dir", "results")
+        output_dir = os.path.join(root_output_dir, f"{name}-{version}")
+        mconfig["dirs"]["_parent"] = output_dir
+    mconfig["dirs"]["_parent"] = mconfig["dirs"]["_parent"].rstrip("/") + "/"
+    os.makedirs(mconfig["dirs"]["_parent"], exist_ok=True)
+
+    # Update paths to conda environments to be relative to the module directory
+    for env_name, env_val in mconfig["conda_envs"].items():
+        if env_val is not None:
+            mconfig["conda_envs"][env_name] = os.path.realpath(env_val)
+
+    # Setup output sub-directories
+    scratch_subdirs = mconfig.get("scratch_subdirectories", [])
+    mconfig = op.setup_subdirs(mconfig, subdirectories, scratch_subdirs)
+
+    # Setup log sub-directories
+    mconfig["logs"] = dict()
+    parent_dir = mconfig["dirs"]["_parent"]
+    launched_fmt = op._session.launched_fmt
+    logs_parent_dir = os.path.join(parent_dir, "logs", launched_fmt)
+    logs_parent_dir = logs_parent_dir.rstrip("/") + "/"
+    mconfig["logs"]["_parent"] = logs_parent_dir
+    os.makedirs(logs_parent_dir, exist_ok=True)
+    for subdir, value in mconfig["dirs"].items():
+        if subdir == "_parent":
+            continue
+        mconfig["logs"][subdir] = value.replace(parent_dir, logs_parent_dir)
+
+    # Generate runs
+    assert "pairing_config" in mconfig, "Module config must have 'pairing_config'."
+    runs = op.generate_runs(
+        msamples, mconfig["pairing_config"], mconfig.get("unmatched_normal_ids")
+    )
+
+    # Split runs based on pair_status
+    mconfig["runs"] = runs
+    mconfig["paired_runs"] = runs[runs.pair_status != "no_normal"]
+    mconfig["unpaired_runs"] = runs[runs.pair_status == "no_normal"]
+
+    # Return module-specific configuration
+    config["lcr-modules"][name] = mconfig
+    return mconfig
+
+
+# Setup module and store module-specific configuration in `CFG_SLMS3`
+# `CFG_SLMS3` is a shortcut to `config["lcr-modules"]["slms_3"]`
+# This is not assigned to `CFG` as in other modules because the sub-modules here will use `CFG`. 
+CFG_SLMS3 = setup_module(
+    name = "slms_3",
+    version = "1.0",
+    subdirectories = ["inputs", "strelka_gnomad", "lofreq_gnomad", "strelka_lofreq_union", "sage_gnomad", "mutect2_depth_filt", "union", "outputs"],
+)
+
+# Ensure each of the submodule versions meets requirements. 
+
```

Comment:
Seems like a handy feature we might want to roll right into oncopipe eventually!